src/drivers/codemodel-driver-interface.ts

```diff
@@ -0,0 +1,118 @@
+import {CMakeDriver} from '@cmt/drivers/driver';
+import * as vscode from 'vscode';
+
+/**
+ * This file contains the API description between IDE parts and the cmake model driver.
+ * This API CodeModel contains only current required cmake code model parts.
+ * There are more information provided by CMake as mapped.
+ */
+
+/**
+ * Describes all required methods for access to the build code model of the driver
+ */
+export abstract class CodeModelDriver extends CMakeDriver {
+  /**
+   * Event registration for code model updates
+   *
+   * This event is fired after update of the code model, like after cmake configuration.
+   */
+  abstract onCodeModelChanged: vscode.Event<CodeModelContent|null>;
+}
+
+export type TargetTypeString
+    = ('STATIC_LIBRARY'|'MODULE_LIBRARY'|'SHARED_LIBRARY'|'OBJECT_LIBRARY'|'EXECUTABLE'|'UTILITY'|'INTERFACE_LIBRARY');
+
+/** Describes a cmake target */
+export interface CodeModelTarget {
+  /**
+   * A string specifying the logical name of the target.
+   *
+   * (Source CMake Documentation cmake-file-api(7))
+   */
+  readonly name: string;
+
+  /**
+   * A string specifying the type of the target.
+   * The value is one of EXECUTABLE, STATIC_LIBRARY, SHARED_LIBRARY, MODULE_LIBRARY, OBJECT_LIBRARY, or UTILITY.
+   *
+   * (Source CMake Documentation cmake-file-api(7))
+   *
+   * \todo clarify need of INTERFACE_LIBRARY type
+   */
+  type: TargetTypeString;
+
+  /** A string specifying the absolute path to the target’s source directory. */
+  sourceDirectory?: string;
+
+  /** Name of the target artefact on disk (library or executable file name). */
+  fullName?: string;
+
+  /** List of absolute path of a target build artifact. */
+  artifacts?: string[];
+
+  /**
+   * List of compilation information for artifacts of this target.
+   * It contains groups of source files which there compilation information.
```

Comment:
"It contains groups of source files for which there is compilation information."